numba/targets/builtins.py

```diff
@@ -446,22 +446,48 @@ def impl_index_value(context, builder, sig, args):
     index_value.value = value
     return index_value._getvalue()
 
+
 @overload(min)
-def indval_min(indval1, indval2):
+def indval_min(indval1, indval2=None):
+
     if isinstance(indval1, IndexValueType) and \
        isinstance(indval2, IndexValueType):
-        def min_impl(indval1, indval2):
+        def impl(indval1, indval2=None):
             if indval1.value > indval2.value:
                 return indval2
             return indval1
-        return min_impl
+        return impl
+
+    if indval2 is None:
+        if isinstance(indval1, types.IterableType):
```

Comment:
Wondering if it's possible to pull these out of this scope which has a non-intuitive signature? Would defining another overload like:
```python
@overload(min)
def iterable_min(iterable):
    if isinstance(iterable, types.IterableType):
        def impl(iterable):
            <details>
```
work? Could also close over the `<`, `>` operator to save duplication too?

Reply:
Done.